pkg/client/events/replay_client.go

```diff
@@ -210,15 +194,30 @@ func (rClient *replayClient[T]) goPublishEvents(ctx context.Context, publishCh c
 		select {
 		case <-ctx.Done():
 			// If the context is done, exit the loop and stop processing events.
+			rClient.logger.Info().Msgf(
+				"🛑 Context canceled. Stopping event subscription for query: %s",
+				rClient.queryString,
+			)
 			return
 		default:
+			// Log connection attempt state
+			if rClient.currentState.status != "connected" {
```

Comment:
Can you move these "magic strings" into local constants or turn them into iotas?